rs/bitcoin/kyt/src/main.rs

```diff
@@ -127,8 +146,103 @@ fn post_upgrade(arg: KytArg) {
+                ic_cdk::api::stable::stable_size() as f64 * 65536.0,
+                "Size of the stable memory allocated by this canister.",
+            )
+            .unwrap();
+
+        STATS.with(|s| {
+            let stats = s.borrow();
+            let mut counter = writer
+                .counter_vec(
+                    "btc_kyt_http_calls_total",
+                    "The number of http outcalls made since the last canister upgrade.",
+                )
+                .unwrap();
+            for (status, count) in stats.https_outcall_status.iter() {
+                counter = counter
+                    .value(&[("status", status.to_string().as_str())], *count as f64)
+                    .unwrap();
+            }
+            writer
+                .counter_vec(
+                    "ckbtc_kyt_requests_total",
+                    "The number of KYT requests received since the last canister upgrade.",
+                )
+                .unwrap()
+                .value(
+                    &[("type", "check_transaction")],
+                    stats.check_transaction_count as f64,
+                )
+                .unwrap()
+                .value(
+                    &[("type", "check_address")],
+                    stats.check_address_count as f64,
+                )
+                .unwrap();
+        });
+
+        http::HttpResponseBuilder::ok()
+            .header("Content-Type", "text/plain; version=0.0.4")
+            .with_body_and_content_length(writer.into_inner())
+            .build()
+    } else if req.path() == "/logs" {
+        use serde_json;
+
+        let max_skip_timestamp = match req.raw_query_param("time") {
+            Some(arg) => match u64::from_str(arg) {
+                Ok(value) => value,
+                Err(_) => {
+                    return http::HttpResponseBuilder::bad_request()
+                        .with_body_and_content_length("failed to parse the 'time' parameter")
+                        .build()
+                }
+            },
+            None => 0,
+        };
+
+        let mut entries: Log = Default::default();
+        for entry in export_logs(&P0) {
+            entries.entries.push(LogEntry {
+                timestamp: entry.timestamp,
+                counter: entry.counter,
+                priority: Priority::P0,
+                file: entry.file.to_string(),
+                line: entry.line,
+                message: entry.message,
+            });
+        }
+        for entry in export_logs(&P1) {
+            entries.entries.push(LogEntry {
+                timestamp: entry.timestamp,
+                counter: entry.counter,
+                priority: Priority::P1,
+                file: entry.file.to_string(),
+                line: entry.line,
+                message: entry.message,
+            });
+        }
+        entries
+            .entries
+            .retain(|entry| entry.timestamp >= max_skip_timestamp);
```

Comment:
nit: the filter could be applied directly when building `entries` to avoid storing potentially a large number of events that are not needed.